### PR TITLE
playlist_id should be case-sensitive #8138

### LINKED ISSUE #8138: [media.ccc.de:lists] playlist_id should be case-sensitive
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Provide a description that is worded well enough to be understood

Some playlists use uppercase `playlist_id`s, like `https://media.ccc.de/c/DS2023` → `https://media.ccc.de/public/conferences/DS2022` or `https://media.ccc.de/c/MCH2022` → `https://media.ccc.de/public/conferences/MCH2022`. So I guess removing `.lower()` in https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/ccc.py#L96 should resolve this.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ python -m yt_dlp --verbose --ignore-config https://media.ccc.de/c/DS2023
[debug] Command-line config: ['--verbose', '--ignore-config', 'https://media.ccc.de/c/DS2023']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (source)
[debug] Lazy loading extractors is disabled
[debug] Git HEAD: 30ba233d4
[debug] Python 3.11.5 (CPython x86_64 64bit) - Linux-6.5.0-5-generic-x86_64-with-glibc2.38 (OpenSSL 3.0.10 1 Aug 2023, glibc 2.38)
[debug] exe versions: ffmpeg 6.0 (setts), ffprobe 6.0, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.46.0, pyxattr-0.8.1, secretstorage-3.3.3, sqlite3-2.6.0, websockets-10.4
[debug] Proxy map: {}
[debug] Loaded 1866 extractors
[media.ccc.de:lists] Extracting URL: https://media.ccc.de/c/DS2023
[media.ccc.de:lists] ds2023: Downloading JSON metadata
ERROR: [media.ccc.de:lists] DS2023: Unable to download JSON metadata: HTTP Error 404: Not Found (caused by <HTTPError 404: Not Found>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[…]
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/tmp/yt-dlp/yt_dlp/extractor/common.py", line 847, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/yt-dlp/yt_dlp/YoutubeDL.py", line 4078, in urlopen
    raise _CompatHTTPError(e) from e
yt_dlp.networking.exceptions._CompatHTTPError: HTTP Error 404: Not Found
```

### PR DESCRIPTION
**Pull request by Rohxn16**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Playlist Id is now case sensitive.
Console output : 
`./yt-dlp.sh --verbose --ignore-config https://media.ccc.de/c/DS2023
[debug] Command-line config: ['--verbose', '--ignore-config', 'https://media.ccc.de/c/DS2023']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (source)
[debug] Git HEAD: 20fbbd924
[debug] Python 3.9.2 (CPython x86_64 64bit) - Linux-6.1.0-1parrot1-amd64-x86_64-with-glibc2.31 (OpenSSL 1.1.1n  15 Mar 2022, glibc 2.31)
[debug] exe versions: none
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1866 extractors
[media.ccc.de:lists] Extracting URL: https://media.ccc.de/c/DS2023
[media.ccc.de:lists] DS2023: Downloading JSON metadata
[download] Downloading playlist: Datenspuren 2023
[media.ccc.de:lists] Playlist Datenspuren 2023: Downloading 37 items of 37
[download] Downloading item 1 of 37`

Fixes #8138 

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🐛🆔🔤

Fix playlist download bug in `ccc.py` extractor. Preserve the original case of the playlist ID instead of lowercasing it.

> _`playlist_id` fixed_
> _No more lowercasing bugs_
> _Winter downloads work_

### Walkthrough
* Fix playlist download bug by removing `.lower()` from `playlist_id` ([link](https://github.com/yt-dlp/yt-dlp/pull/8144/files?diff=unified&w=0#diff-f51824075dfb87d4a82e37e0fcf75eee34a78c52a265cceb6f9fd2f2b3fdb05cL96-R96))

</details>

### CODE REVIEW

**Comment by bashonly** [comment]
Add https://media.ccc.de/c/DS2023 as a test

**Comment by Rohxn16** [comment]
i hope this does the job. This is my first ever commit, i am sorry if there are any mistakes in my work.

**Comment by bashonly** [comment]
The change is exactly right, we just need a test added to `_TESTS`

**Comment by Rohxn16** [comment]
Understood. Thank you.

**Comment by Rohxn16** [comment]
class CCCPlaylistIE(InfoExtractor):
    IE_NAME = 'media.ccc.de:lists'
    _VALID_URL = r'https?://(?:www\.)?media\.ccc\.de/c/(?P<id>[^/?#&]+)'
    _TESTS = [{
        'url': 'https://media.ccc.de/c/30c3',
        'info_dict': {
            'title': '30C3',
            'id': '30c3',
        },
        'playlist_count': 135,
    }]

So, is this where I need to add the test case? I am sorry if i am asking dumb questions.

**Comment by bashonly** [comment]
Yeah, add a test dict to the `_TESTS` list. You could just duplicate the already-existing test and then change its `url`, `id` and `title` values. Adjust the `playlist_count` value so that it is correct and the test passes

Run the test like this
`python test/test_download.py TestDownload.test_CCCPlaylist_1`

File: yt_dlp/extractor/ccc.py [hunk 1]
```diff
@@ -90,10 +90,17 @@ class CCCPlaylistIE(InfoExtractor):
             'id': '30c3',
         },
         'playlist_count': 135,
+    },{
+        'url':"https://media.ccc.de/c/DS2023",
+        'info_dict':{
+            'title':'Datenspuren 2023',
+            'id':'DS2023',
+        },
+        'playlist_count':37
```
Comment (bashonly) [inline yt_dlp/extractor/ccc.py:99]:
```suggestion
    }, {
        'url': 'https://media.ccc.de/c/DS2023',
        'info_dict': {
            'title': 'Datenspuren 2023',
            'id': 'DS2023',
        },
        'playlist_count': 37,
```

File: yt_dlp/extractor/ccc.py [hunk 2]
```diff
@@ -90,10 +90,17 @@ class CCCPlaylistIE(InfoExtractor):
             'id': '30c3',
         },
         'playlist_count': 135,
+    }, {
+        'url': "https://media.ccc.de/c/DS2023",
```
Comment (bashonly) [inline yt_dlp/extractor/ccc.py:94]: single quotes
```suggestion
        'url': 'https://media.ccc.de/c/DS2023',
```

**Review by bashonly — APPROVED** [review]

**Comment by Rohxn16** [comment]
I think that does it. Thank you for bearing with my silly mistakes.

**Comment by bashonly** [comment]
No worries, thanks for the PR

**Pull request merged**